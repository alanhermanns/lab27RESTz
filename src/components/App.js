import React, { Component } from 'react';
import Header from './Header';
import HeadersForm from './HeadersForm';
import FormURLAndMethods from './FormURL';
import Results from './Results';
import hitApi from '../services/hitApi';
import History from './History';

export default class App extends Component {

  state = {
    URL: '',
    Method:'get',
    loading: true,
    pastRequests: [],
    resultJSON: []
  }

    fetch = () => {
      const {URL, Method, BODY} = this.state;
      if(!this.state.BODY){
        fetch(URL, {
          'method': Method
        })
        .then(res => {
          return res.json()
        })
        .then(response => {
          this.setState({resultJSON : [JSON.stringify(response)]})
        })
      }
      else fetch(URL, {
        'method': Method,
        'body': JSON.stringify(BODY)
      })
      .then(res => {
        return res.json()
      })
      .then(response => {
        this.setState({resultJSON : [JSON.stringify(response)]})
      })
    }

    URLHandler = event => {
      event.preventDefault();
      let URL = this.state.URL;
      this.setState(state => ({pastRequests: [...state.pastRequests, URL]}))
      console.log(URL)
      this.fetch();
    }
    methodHandler= (event) => {
      event.preventDefault()
      let value = event.target.value;
      this.setState(state => ({Method: value}))
    }
    sendTextToState = (event) => {
      let value = event.target.value;
      this.setState(state => ({URL : value}))
      console.log(this.state.URL)
    }
    sendBodyToState = (event) => {
      let value = event.target.value;
      this.setState(state => ({BODY : value}))
    }

    render()
    {
    return <>
    <Header />
    <FormURLAndMethods URLHandler={this.URLHandler} methodHandler={this.methodHandler} sendTextToState={this.sendTextToState} sendBodyToState={this.sendBodyToState}/>
    <HeadersForm />
    <History history={this.state.pastRequests}/>
    <Results someJson={this.state.resultJSON}/>
    </>
    }
}
  