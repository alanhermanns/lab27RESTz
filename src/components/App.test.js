import React from 'react';
import { shallow } from 'enzyme';
import App from './App';



describe('App component', () => {
  it('has working methods', () => {
    const wrapper = shallow(<App />);
    const app = wrapper.instance();
    app.methodHandler({
      target: {
        value: 'post'
      },
      preventDefault: () => {
        return 'hello'
      }
    })
    app.sendTextToState({
      target: {
        value: 'Text'
      }
    })
    app.sendBodyToState({
      target: {
        value: 'body'
      }
    })

    expect(app.state).toEqual({
    BODY: 'body',
    URL: 'Text',
    Method:'post',
    loading: true,
    pastRequests: [],
    resultJSON: []
    })
  });
});
  