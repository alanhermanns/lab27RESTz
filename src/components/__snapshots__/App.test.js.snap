// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
  <Fragment>
    <Header />
    <FormURLAndMethods
      URLHandler={[Function]}
      methodHandler={[Function]}
      sendBodyToState={[Function]}
      sendTextToState={[Function]}
    />
    <FormURLAndMethods />
    <History
      history={Array []}
    />
    <Results
      someJson={Array []}
    />
  </Fragment>
`;
